language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/hover/HoverObjectResolver.java

```diff
@@ -159,11 +164,42 @@ private Hover getHoverObjectForSymbol(FunctionSymbol functionSymbol) {
         }
         List<String> hoverContent = new ArrayList<>();
         documentation.get().description().ifPresent(hoverContent::add);
-
+        List<PathParameterSymbol> parameterSymbols = new ArrayList<>();
+        boolean isResourceMethod = functionSymbol.kind() == SymbolKind.RESOURCE_METHOD;
+
+        if (isResourceMethod) {
+            ResourcePath resourcePath = ((ResourceMethodSymbol) functionSymbol).resourcePath();
+            if (resourcePath.kind() == ResourcePath.Kind.PATH_SEGMENT_LIST) {
+                PathSegmentList pathSegmentList = (PathSegmentList) resourcePath;
+                List<PathParameterSymbol> pathParameterSymbols = pathSegmentList.pathParameters();
+                parameterSymbols.addAll(pathParameterSymbols);
+                if (pathSegmentList.pathRestParameter().isPresent()) {
+                    parameterSymbols.add(pathSegmentList.pathRestParameter().get());
+                }
+            } else if (resourcePath.kind() == ResourcePath.Kind.PATH_REST_PARAM) {
+                parameterSymbols.add(((PathRestParam) resourcePath).parameter());
```

Comment:
Please check whether why isn't this covered by tests? Check the other place as well.